packages/core/theme/core-theme-browser-internal/src/core_theme_provider.tsx

```diff
@@ -6,55 +6,12 @@
  * Side Public License, v 1.
  */
 
-import React, { FC, useMemo } from 'react';
-import { Observable } from 'rxjs';
-import useObservable from 'react-use/lib/useObservable';
-import createCache from '@emotion/cache';
-import { EuiProvider } from '@elastic/eui';
-import { EUI_STYLES_GLOBAL } from '@kbn/core-base-common';
-import type { CoreTheme } from '@kbn/core-theme-browser';
-import { convertCoreTheme } from './convert_core_theme';
-
-const defaultTheme: CoreTheme = {
-  darkMode: false,
-};
-
-interface CoreThemeProviderProps {
-  theme$: Observable<CoreTheme>;
-  globalStyles?: boolean;
-}
-
-const globalCache = createCache({
-  key: 'eui',
-  container: document.querySelector(`meta[name="${EUI_STYLES_GLOBAL}"]`) as HTMLElement,
-});
-const emotionCache = createCache({
-  key: 'css',
-  container: document.querySelector(`meta[name="emotion"]`) as HTMLElement,
-});
-emotionCache.compat = true;
+import { KibanaThemeProvider } from '@kbn/react-kibana-context';
 
 /**
  * Wrapper around `EuiProvider` converting (and exposing) core's theme to EUI theme.
  * @internal Only meant to be used within core for internal usages of EUI/React
+ * @deprecated use `KibanaThemeProvider` from `@kbn/react-kibana-context
  */
-export const CoreThemeProvider: FC<CoreThemeProviderProps> = ({
-  theme$,
-  children,
-  globalStyles,
-}) => {
-  const coreTheme = useObservable(theme$, defaultTheme);
-  const euiTheme = useMemo(() => convertCoreTheme(coreTheme), [coreTheme]);
-  const includeGlobalStyles = globalStyles === false ? false : undefined;
-  return (
-    <EuiProvider
-      globalStyles={includeGlobalStyles}
-      utilityClasses={includeGlobalStyles}
-      colorMode={euiTheme.colorMode}
-      theme={euiTheme.euiThemeSystem}
-      cache={{ default: emotionCache, global: globalCache }}
-    >
-      {children}
-    </EuiProvider>
-  );
-};
+export const CoreThemeProvider = KibanaThemeProvider;
+CoreThemeProvider.displayName = 'CoreThemeProvider';
```

Comment:
The original instance will also get updated. Or is that intended?

```suggestion
export const CoreThemeProvider = KibanaThemeProvider.bind(undefined);
CoreThemeProvider.displayName = 'CoreThemeProvider';
```

Reply:
FFR, this doesn't work.  You have to use `Object.assign`:

```
export const CoreThemeProvider = Object.assign(KibanaThemeProvider, {
  displayName: 'CoreThemeProvider',
});
```